examples/vpc-endpoints/main.tf

```diff
@@ -0,0 +1,117 @@
+provider "aws" {
+  region = var.region
+}
+
+locals {
+  gateway_vpc_endpoints = {
+    "s3" = {
+      name = "s3"
+      policy = jsonencode({
+        Version = "2012-10-17"
+        Statement = [
+          {
+            Action = [
+              "s3:*",
+            ]
+            Effect    = "Allow"
+            Principal = "*"
+            Resource  = "*"
+          },
+        ]
+      })
+    }
+    "dynamodb" = {
+      name   = "dynamodb"
+      policy = null
+    }
+  }
+  interface_vpc_endpoints = {
+    "ec2" = {
+      name                = "ec2"
+      security_group_ids  = [aws_security_group.ec2_vpc_endpoint_sg.id]
+      subnet_ids          = [module.subnets.private_subnet_ids[0]]
```

Comment:
why are we providing only one subnet even if the submodule takes a list?

Reply:
There is a limitation that there is a maximum of one subnet per AZ for which the Interface VPC Endpoint is deployed to. I've made this simple for example sake. Technically, the Interface VPC endpoint only needs to deploy an ENI to one AZ. Unless it is to be HA.

Reply:
Nevermind, I think this is something that affected the example during early development, and before switching the testing region to `us-east-2`. I've fixed this now.